Extension fails to start on vscode 1.96.4

Problems started quite recently, but I'm not sure exactly when.

```
console.ts:137 [Extension Host] rejected promise not handled within 1 second: Error: illegal argument `name`. must not be falsy
EYi @ console.ts:137
console.ts:137 [Extension Host] stack trace: Error: illegal argument `name`. must not be falsy
    at Lw.createOutputChannel (file:///vscode/vscode-server/bin/linux-x64/cd4ee3b1c348a13bafd8f9ad8060705f6d4b9cba/out/vs/workbench/api/node/extensionHostProcess.js:121:894)
    at Object.createOutputChannel (file:///vscode/vscode-server/bin/linux-x64/cd4ee3b1c348a13bafd8f9ad8060705f6d4b9cba/out/vs/workbench/api/node/extensionHostProcess.js:157:48012)
    at o.get outputChannel [as outputChannel] (/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:156845)
    at o.logOutputMessage (/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:161984)
    at o.error (/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:161889)
    at o.start (/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:165948)
EYi @ console.ts:137
mainThreadExtensionService.ts:78 [pgourlain.erlang]illegal argument `name`. must not be falsy
$onExtensionRuntimeError @ mainThreadExtensionService.ts:78
mainThreadExtensionService.ts:79 Error: illegal argument `name`. must not be falsy
    at Lw.createOutputChannel (/vscode/vscode-server/bin/linux-x64/cd4ee3b1c348a13bafd8f9ad8060705f6d4b9cba/out/vs/workbench/api/node/extensionHostProcess.js:121:894)
    at Object.createOutputChannel (/vscode/vscode-server/bin/linux-x64/cd4ee3b1c348a13bafd8f9ad8060705f6d4b9cba/out/vs/workbench/api/node/extensionHostProcess.js:157:48012)
    at o.get outputChannel [as outputChannel] (vscode-file://vscode-app/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:156845)
    at o.logOutputMessage (vscode-file://vscode-app/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:161984)
    at o.error (vscode-file://vscode-app/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:161889)
    at o.start (vscode-file://vscode-app/home/vscode/.vscode-server/extensions/pgourlain.erlang-1.1.1/out/lib/extension.js:1:165948)
$onExtensionRuntimeError @ mainThreadExtensionService.ts:79
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change log
 
+## Version 1.1.2 (February 22, 2025)
+
+* [320](https://github.com/pgourlain/vscode_erlang/issues/320) : Extension fails to start on vscode 1.96.4.
+* [321](https://github.com/pgourlain/vscode_erlang/issues/321) : Remove warnings.
+
+
 ## Version 1.1.1 (December 8, 2024)
 
 * [319](https://github.com/pgourlain/vscode_erlang/issues/319) : Add a configuration setting to change the storage mode for large cache tables (mainly used for code navigation). 

--- a/apps/erlangbridge/src/gen_lsp_doc_server.erl
+++ b/apps/erlangbridge/src/gen_lsp_doc_server.erl
@@ -469,11 +469,10 @@ safe_new_table(Name, dets, Type, ExtraCreateOpts) ->
 
 delete_cache_file(Name) ->
     case dets:info(Name, filename) of
+        undefined -> ok;
         FileName ->
             dets:close(Name),
-            file:delete(FileName);
-        _ ->
-            ok
+            file:delete(FileName)
     end.
 
 parse_and_store(File, ContentsFile) ->

--- a/apps/erlangbridge/src/lsp_syntax.erl
+++ b/apps/erlangbridge/src/lsp_syntax.erl
@@ -119,69 +119,69 @@ lint(FileSyntaxTree, File) ->
 	        #{parse_result => false, error_message => <<"lint error">>}
     end.
 
-combine_lint(LintResult, SyntaxTree, File) ->
-    case LintResult of
-        #{parse_result := ParseResult, errors_warnings := ErrorsWarnings} 
-            when ParseResult =:= true -> 
-              % combine only if previous parse is successfull
-              #{
-                parse_result => ParseResult, 
-                errors_warnings => lsp_lint(ErrorsWarnings, SyntaxTree, File)
-            };
-        _ -> LintResult
-    end.
+% combine_lint(LintResult, SyntaxTree, File) ->
+%     case LintResult of
+%         #{parse_result := ParseResult, errors_warnings := ErrorsWarnings} 
+%             when ParseResult =:= true -> 
+%               % combine only if previous parse is successfull
+%               #{
+%                 parse_result => ParseResult, 
+%                 errors_warnings => lsp_lint(ErrorsWarnings, SyntaxTree, File)
+%             };
+%         _ -> LintResult
+%     end.
 
-lsp_lint(PreviousLintResult, SyntaxTree, File) ->
-    RootWorkspace = gen_lsp_config_server:root(),
-    %%process remote call
-    fold_in_syntax_tree(fun
-        ({call, {_, _}, {remote, {_, _}, {atom, {Line, Column}, FnModule}, {atom, {_, _}, FnName}}, Args}, _CurrentFile, Acc)
-            -> 
-                % Incr = 1 for last right parenthesis
-                Acc ++ check_if_remote_fun_exists(RootWorkspace, FnModule, FnName, length(Args), range_of(Line, Column, FnModule, FnName, Args, 1));
-        ({'fun', {_, _}, {function, {atom, {Line, Column}, FnModule}, {atom, {_,_}, FnName}, {integer, {_, End}, FnArity}}}, _CurrentFile, Acc)
-             ->
-            Width = length(lsp_utils:to_string("~w",[FnArity])),
-            Acc ++ check_if_remote_fun_exists(RootWorkspace, FnModule, FnName, FnArity,{Line, Column, Line, End + Width});
+% lsp_lint(PreviousLintResult, SyntaxTree, File) ->
+%     RootWorkspace = gen_lsp_config_server:root(),
+%     %%process remote call
+%     fold_in_syntax_tree(fun
+%         ({call, {_, _}, {remote, {_, _}, {atom, {Line, Column}, FnModule}, {atom, {_, _}, FnName}}, Args}, _CurrentFile, Acc)
+%             -> 
+%                 % Incr = 1 for last right parenthesis
+%                 Acc ++ check_if_remote_fun_exists(RootWorkspace, FnModule, FnName, length(Args), range_of(Line, Column, FnModule, FnName, Args, 1));
+%         ({'fun', {_, _}, {function, {atom, {Line, Column}, FnModule}, {atom, {_,_}, FnName}, {integer, {_, End}, FnArity}}}, _CurrentFile, Acc)
+%              ->
+%             Width = length(lsp_utils:to_string("~w",[FnArity])),
+%             Acc ++ check_if_remote_fun_exists(RootWorkspace, FnModule, FnName, FnArity,{Line, Column, Line, End + Width});
 
-    (_, _, Acc) -> Acc
-    end,
-    PreviousLintResult, File, SyntaxTree).
+%     (_, _, Acc) -> Acc
+%     end,
+%     PreviousLintResult, File, SyntaxTree).
 
-range_of(Line, Column, MName, FName, Args, Incr) ->
-    {L,C, NewIncr} = if 
-        length(Args) > 0 -> max_lc_root(lists:last(Args), {Line, Column, Incr});
-        true -> 
-            % +2 for parenthesis
-            {Line, Column, Incr + 2 + length(atom_to_list(MName)) +length(atom_to_list(FName))}
-    end,
-   {Line, Column, L, C+NewIncr}.
+% range_of(Line, Column, MName, FName, Args, Incr) ->
+%     {L,C, NewIncr} = if 
+%         length(Args) > 0 -> max_lc_root(lists:last(Args), {Line, Column, Incr});
+%         true -> 
+%             % +2 for parenthesis
+%             {Line, Column, Incr + 2 + length(atom_to_list(MName)) +length(atom_to_list(FName))}
+%     end,
+%    {Line, Column, L, C+NewIncr}.
 
 
-max_lc_root({_,_,Atom}=Item, {Line, Column, Incr}) when is_atom(Atom) ->
-    max_lc(Item, {Line, Column, Incr + length(atom_to_list(Atom))});
-max_lc_root({_,_,Int}=Item, {Line, Column, Incr}) when is_number(Int) ->
-    ArgLength = length(lsp_utils:to_string("~w",[Int])),
-    max_lc(Item, {Line, Column, Incr + ArgLength});
-max_lc_root(Item, Acc) ->
-    max_lc(Item, Acc).
+% max_lc_root({_,_,Atom}=Item, {Line, Column, Incr}) when is_atom(Atom) ->
+%     max_lc(Item, {Line, Column, Incr + length(atom_to_list(Atom))});
+% max_lc_root({_,_,Int}=Item, {Line, Column, Incr}) when is_number(Int) ->
+%     ArgLength = length(lsp_utils:to_string("~w",[Int])),
+%     max_lc(Item, {Line, Column, Incr + ArgLength});
+% max_lc_root(Item, Acc) ->
+%     max_lc(Item, Acc).
 
--spec max_lc(Item:: term(), Acc :: term()) -> Acc1 :: term().
-max_lc({_, {L,C}, _}, {CL, CC, Incr}) when L > CL orelse (L =:= CL andalso C >= CC) ->
-    {L, C, Incr};
-max_lc({_, {_,_}, Item}, Acc) ->
-    max_lc(Item, Acc);
-max_lc({call, {L,C}, Remote, Args}, {CL, CC, Incr}) when L > CL orelse (L =:= CL andalso C >= CC) ->
-    if 
-        length(Args) > 0 -> max_lc_root(lists:last(Args), {L, C, Incr+1});
-        true -> 
-            case Remote of
-                {remote, _, {atom, _, _}, {atom,{RL, RC}, Atom}} -> {RL, RC, Incr + 1 + length(atom_to_list(Atom))};
-                _ -> {L, C, Incr}
-            end            
-    end;
-max_lc(_, Acc) -> 
-    Acc.
+% -spec max_lc(Item:: term(), Acc :: term()) -> Acc1 :: term().
+% max_lc({_, {L,C}, _}, {CL, CC, Incr}) when L > CL orelse (L =:= CL andalso C >= CC) ->
+%     {L, C, Incr};
+% max_lc({_, {_,_}, Item}, Acc) ->
+%     max_lc(Item, Acc);
+% max_lc({call, {L,C}, Remote, Args}, {CL, CC, Incr}) when L > CL orelse (L =:= CL andalso C >= CC) ->
+%     if 
+%         length(Args) > 0 -> max_lc_root(lists:last(Args), {L, C, Incr+1});
+%         true -> 
+%             case Remote of
+%                 {remote, _, {atom, _, _}, {atom,{RL, RC}, Atom}} -> {RL, RC, Incr + 1 + length(atom_to_list(Atom))};
+%                 _ -> {L, C, Incr}
+%             end            
+%     end;
+% max_lc(_, Acc) -> 
+%     Acc.
 
 check_if_remote_fun_exists(RootWorkspace, FnModule, FnName, FnArity, {Line, Column, LE,LC}) ->
     % check if module is under workspace

--- a/lib/RebarRunner.ts
+++ b/lib/RebarRunner.ts
@@ -232,7 +232,7 @@ export class RebarRunner implements vscode.Disposable {
 
 	public static get RebarOutput(): vscode.OutputChannel {
 		if (!rebarOutputChannel) {
-			rebarOutputChannel = vscode.window.createOutputChannel('rebar');
+			rebarOutputChannel = vscode.window.createOutputChannel('rebar', 'erlang');
 		}
 		return rebarOutputChannel;
 	}

--- a/lib/lsp/lspclientextension.ts
+++ b/lib/lsp/lspclientextension.ts
@@ -204,7 +204,7 @@ function getPort(callback) {
 export function activate(context: ExtensionContext) {
 	let erlangCfg = getElangConfigConfiguration();
 	if (erlangCfg.verbose)
-		lspOutputChannel = Window.createOutputChannel('Erlang Language Server');
+		lspOutputChannel = Window.createOutputChannel('Erlang Language Server', 'erlang');
 
 	lspValue.activate(context, lspOutputChannel);
 	lspRename.activate(context, lspOutputChannel);

--- a/lib/vscodeAdapter.ts
+++ b/lib/vscodeAdapter.ts
@@ -6,7 +6,7 @@ var erlangOutputChannel : vscode.OutputChannel;
 
 export function ErlangOutput() : vscode.OutputChannel {
     if (!erlangOutputChannel) {
-        erlangOutputChannel = vscode.window.createOutputChannel('erlang'); 
+        erlangOutputChannel = vscode.window.createOutputChannel('erlang', 'erlang'); 
     }
     return erlangOutputChannel;
 }

